nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -40,10 +41,18 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
 
   private def getInts(): Seq[Int] = getSeq(getInt)
 
-  private def getStrings(): Seq[String] = getSeq(getString)
   private def getString(): String = {
-    new String(getBytes(), "UTF-8")
+    if (prelude.revision < 7) getASCIIString()
+    else {
+      val chars = Array.fill(getInt)(getChar)
+      new String(chars)
+    }
+  }
+
+  private def getASCIIString(): String = {
```

Comment:
This should definitely be called `getUTF8String()` if it reads in UTF-8. ASCII is another charset.